sharing/Readme_sharing.md

```diff
@@ -0,0 +1,22 @@
+# Aim/Objective:
+
+The purpose of the ODM is to support wastewater-based surveillance and epidemiology by facilitating the collection, standardization, and transparency of data by providing a more harmonized ODM data format to share data between different data curators and data repositories. 
+
+The end objective of data sharing is to have a function that can take in the user inputs in the form of CSV schema with predefined rules and the user data and filter the data based on the rules in the CSV schema and output back an excel spreadsheet to the user with filtered data that is ready to be used with specific organizations that user wants to share their data with. This is implement using python code.
```

Comment:
At this point, we don't need to specify data as an excel spreadsheet. We may not have been explicit but the default data storage is a CSV and we also have an SQL schema.

Reply:
I have made the change. I removed CSV from schema and excel spreadsheets for output data. But I will need to go over this as my initial thoughts on this were that user gave us excel data as that is what they are mostly using and CSV schema is what we structure for them to document their rules just as in case of validate.csv schema file. I structured the code accordingly too without hard coding. I have made the changes here though and closing the pull request. I believe you will get better understanding when reading the sharing_schema branch as I defined how the schema is implemented there. Thank you.

Reply:
We may need to discuss how we want to implement different functions throughout the ODM, like mapping, validation and sharing. I had envisioned three steps. 

The first step uses a generic import function that brings in users' data - we would support a few different formats including Excel and CSV. This first step imports the data as a python data frame. 

A second step performs a series of functions - mapping, validation, and sharing - on the data frame. 

The third step exports the data back to the data user in the data format of their choice - CSV or Excel. A challenge or discussion is whether we'd want to preserve the user's original Excel table. I'd keep it simple and not try to preserve the user's original table. Indeed, it may be easiest to just return a CSV file.

Reply:
How I envisioned it and sort of started structuring my code also to get some picture of how this may work behind the scenes is that user will give us the schema which I am converting into json dictionary in python and user gives us data. Based on ODM compliant dataset on github, I used excel for it as they can provide different tables as different sheets of same excel workbook which can be read into pandas dataframe in a dictionary format and different tables can be accessed by keys. The user can selectively request to filter and return only specific tables from their workbook NOT all the tables which will be part of the web form they fill.
The code filters the data and then returns it to them as excel workbook with different sheets as different tables they requested but of course the output could be changed to be the multiple CSV files being outputted for different tables.